bitsandbytes/nn/modules.py

```diff
@@ -311,9 +316,13 @@ def cpu(self):
         return self.to(device="cpu")
 
     def cuda(self, device: Optional[int | device | str] = None, non_blocking: bool = False):
+        if getattr(self.quant_state, "packing_format_for_cpu", False):
+            self.data, self.quant_state = _convert_weight_packed_for_cpu_inverse(self.data, self.quant_state)
```

Comment:
I think maybe this is something that would be better done in `to()` but since you make the point that we don't do the avx512 CPU build when you're using CUDA/XPU, it won't really matter yet - shouldn't end up repacking. So I'm OK to come back to it later if we separate CPU code out of the accelerator libs.